### ISSUE TITLE
detect param aliasing with no-param-reassign { "props": true }

### ISSUE THREAD

**Issue by graingert** [eslint/eslint#10454]
**What rule do you want to change?**
no-param-reassign { "props": true }
**Does this change cause the rule to produce more or fewer warnings?**
more

**How will the change be implemented? (New option, new default behavior, etc.)?**
not sure

**Please provide some example code that this change will affect:**
see next two answers

**What does the rule currently do for this code?**
```js
/*eslint no-param-reassign: ["error", { "props": 'true' }]*/
// good
function foo(bar) {
    const sneakyBar = bar;
    sneakyBar.prop = "value";
}
// good
function foo(bar) {
    const sneakyBar = bar;
    delete sneakyBar.aaa;
}
// good
function foo(bar) {
    const sneakyBar = bar;
    sneakyBar.aaa++;
}
```

**What will the rule do after it's changed?**

```js
/*eslint no-param-reassign: ["error", { "props": 'no-alias' }]*/
// bad
function foo(bar) {
    const sneakyBar = bar;
    sneakyBar.prop = "value";
}
// bad
function foo(bar) {
    const sneakyBar = bar;
    delete sneakyBar.aaa;
}
// bad
function foo(bar) {
    const sneakyBar = bar;
    sneakyBar.aaa++;
}
```

**Comment by ilyavolodin** [comment]
We usually tried to stay away from reassignment tracking, because it's not possible to do fully, and very hard to do correctly. I'm not opposed to introducing a new option for this, if it can be implemented in a reasonable fashion.

**Comment by not-an-aardvark** [comment]
Thanks for your interest in improving ESLint. Unfortunately, it looks like this issue didn't get consensus from the team, so I'm closing it. We define [consensus](http://eslint.org/docs/maintainer-guide/issues.html#consensus) as having three :+1:s from team members, as well as a team member willing to champion the proposal. This is a high bar by design -- we can't realistically accept and maintain every feature request in the long term, so we only accept feature requests which are useful enough that there is consensus among the team that they're worth adding.

Since ESLint is pluggable and can load custom rules at runtime, the lack of consensus among the ESLint team doesn't need to be a blocker for you using this in your project, if you'd find it useful. It just means that you would need to [implement the rule](http://eslint.org/docs/developer-guide/working-with-rules) yourself, rather than using a bundled rule that is packaged with ESLint.
